libfoundation/src/foundation-typeinfo.cpp

```diff
@@ -803,6 +805,14 @@ static bool MCCommonHandlerTypeInfoCreate(bool p_is_foreign, const MCHandlerType
     for (index_t i = 0; i < p_field_count; ++i)
     {
 	    __MCAssertIsTypeInfo(p_fields[i].type);
+        
+        if (p_fields[i].mode == kMCHandlerTypeFieldModeVariadic)
+        {
+            p_field_count = i;
```

Comment:
Perhaps it's worth asserting that `p_field_count == i + 1` and `i != 0` as these things are checked at (lc-)compile-time aren't they? Just to ensure we don't break the assumptions about the position of the variadic param internally anywhere.